ur_robot_driver/src/urcl_log_handler.cpp

```diff
@@ -0,0 +1,84 @@
+// Universal Robots A/S and/or its licensors. THE SOURCE CODE IS PROVIDED “AS IS” WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING – BUT NOT LIMITED TO – WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, OR
+// NONINFRINGEMENT. USE OF THE SOURCE CODE IS AT YOUR OWN RISK AND UNIVERSAL ROBOTS A/S AND ITS LICENSORS SHALL, TO THE
+// MAXIMUM EXTENT PERMITTED BY LAW, NOT BE LIABLE FOR ANY ERRORS OR MALICIOUS CODE IN THE SOURCE CODE, ANY THIRD-PARTY
+// CLAIMS, OR ANY OTHER CLAIMS AND DAMAGES, INCLUDING INDIRECT, INCIDENTAL, SPECIAL, CONSEQUENTIAL OR PUNITIVE DAMAGES,
+// OR ANY LOSS OF PROFITS, EXPECTED SAVINGS, OR REVENUES, WHETHER INCURRED DIRECTLY OR INDIRECTLY, OR ANY LOSS OF DATA,
+// USE, GOODWILL, OR OTHER INTANGIBLE LOSSES, RESULTING FROM YOUR USE OF THE SOURCE CODE. You may make copies of the
+// Source Code for use in connection with a Universal Robots or UR+ product, provided that you include (i) an
+// appropriate copyright notice (“©  [the year in which you received the Source Code or the Source Code was first
+// published, e.g. “2021”] Universal Robots A/S and/or its licensors”) along with the capitalized section of this notice
+// in all copies of the Source Code. By using the Source Code, you agree to the above terms. For more information,
+// please contact legal@universal-robots.com.
+
+#include <memory>
+#include <utility>
+
+#include "ur_robot_driver/urcl_log_handler.hpp"
+#include "rclcpp/logging.hpp"
+
+namespace ur_robot_driver
+{
+bool g_registered = false;
+std::unique_ptr<UrclLogHandler> g_log_handler(new UrclLogHandler);
+
+UrclLogHandler::UrclLogHandler() = default;
+
+void UrclLogHandler::log(const char* file, int line, urcl::LogLevel loglevel, const char* message)
+{
+  rcutils_log_location_t location = { "", file, (size_t)line };
+  switch (loglevel) {
+    case urcl::LogLevel::DEBUG:
+      rcutils_log(&location, RCUTILS_LOG_SEVERITY::RCUTILS_LOG_SEVERITY_DEBUG, "UR_Client_Library", "%s", message);
+      break;
+    case urcl::LogLevel::INFO:
+      rcutils_log(&location, RCUTILS_LOG_SEVERITY::RCUTILS_LOG_SEVERITY_INFO, "UR_Client_Library", "%s", message);
+      break;
+    case urcl::LogLevel::WARN:
+      rcutils_log(&location, RCUTILS_LOG_SEVERITY::RCUTILS_LOG_SEVERITY_WARN, "UR_Client_Library", "%s", message);
+      break;
+    case urcl::LogLevel::ERROR:
+      rcutils_log(&location, RCUTILS_LOG_SEVERITY::RCUTILS_LOG_SEVERITY_ERROR, "UR_Client_Library", "%s", message);
+      break;
+    case urcl::LogLevel::FATAL:
+      rcutils_log(&location, RCUTILS_LOG_SEVERITY::RCUTILS_LOG_SEVERITY_FATAL, "UR_Client_Library", "%s", message);
+      break;
+    default:
+      break;
+  }
+}
+
+void registerUrclLogHandler()
+{
+  if (g_registered == false) {
+    // Log level is decided by ROS2 log level
+    urcl::setLogLevel(urcl::LogLevel::DEBUG);
+    urcl::registerLogHandler(std::move(g_log_handler));
```

Comment:
> @urmahp I am OK to merge this. Only I don't see where #107 is solved. Can you point me to those lines?

These lines will set the log level to debug and register the loghandler which means that it will be determined by ROS2 log level which messages will actually be printed. 

Currently the log level in the client library is warning. When the robot connects/disconnects to the driver info messages are logged, currently those messages aren't printed, because the log level is warning. With these changes info messages from the client library will be printed with ROS2. Then we will be able to see when the robot connects/disconnects, which solves #107.